pkg/virt-api/rest/profiler.go

```diff
@@ -117,24 +117,24 @@ func (app *SubresourceAPIApp) getPodsNextPage(cpRequest *v1.ClusterProfilerReque
 }
 
 func podIsReadyComponent(pod *k8sv1.Pod) bool {
+	if pod == nil {
+		return false
+	}
+
 	re, _ := regexp.Compile("^(virt-api-|virt-operator-|virt-handler-|virt-controller-).*")
 	isComponentPod := re.MatchString(pod.Name)
 	// filter out any kubevirt related pod that doesn't have profiling capabilities
 	if !isComponentPod {
 		return false
 	}
 
-	if pod == nil {
-		return false
-	} else if pod.Status.Phase != k8sv1.PodRunning {
+	if pod.Status.Phase != k8sv1.PodRunning || pod.DeletionTimestamp != nil {
 		return false
-	} else if pod.DeletionTimestamp != nil {
-		return false
-	} else {
-		for _, cond := range pod.Status.Conditions {
-			if cond.Type == k8sv1.PodReady && cond.Status == k8sv1.ConditionTrue {
-				return true
-			}
+	}
+
+	for _, cond := range pod.Status.Conditions {
+		if cond.Type == k8sv1.PodReady && cond.Status == k8sv1.ConditionTrue {
```

Comment:
nit: what if `cond.Type == k8sv1.PodRead` but `cond.Status != k8sv1.ConditionTrue` ?

we could return false immediately. Not a must as the condition list is short, and maybe the additional code will just make it more complex.

Reply:
Thanks @nunnatsa!
In theory you're right of course, but I think that in practice the difference would be absolutely negligible. As you said, I'm not even sure it's worth the small code complexity it would add.